File: Programming Fundamentals/Labs And Homeworks/BasicSyntaxExercise/Vacation.java
package BasicSyntaxExercise;

import java.util.Scanner;

public class Vacation {
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        int peopleQuantity = Integer.parseInt(scanner.nextLine());
        String groupName = scanner.nextLine();
        String day = scanner.nextLine();
        double price = 0;
        double sum = 0;
        switch (groupName) {
            case "Students":
                if (day.equals("Friday")) {
                    price = 8.45;
                } else if (day.equals("Saturday")) {
                    price = 9.80;
                } else if (day.equals("Sunday")) {
                    price = 10.46;
                }
                sum = peopleQuantity * price;
                if (peopleQuantity >= 30) {
                    sum = sum - sum * 0.15;
                }
                break;
            case "Business":
                if (day.equals("Friday")) {
                    price = 10.90;
                } else if (day.equals("Saturday")) {
                    price = 15.60;
                } else if (day.equals("Sunday")) {
                    price = 16.0;
                }
                sum = peopleQuantity * price;
                if (peopleQuantity >= 100) {
                    sum = sum - (10 * price);
                }
                break;
            case "Regular":
                if (day.equals("Friday")) {
                    price = 15.0;
                } else if (day.equals("Saturday")) {
                    price = 20.0;
                } else if (day.equals("Sunday")) {
                    price = 22.50;
                }
                sum = peopleQuantity * price;
                if (peopleQuantity >= 10 && peopleQuantity <= 20) {
                    sum = sum - sum * 0.05;
                }
                break;

        }
        System.out.printf("Total price: %.2f", sum);

    }
}

File: JS Front-End/JS Syntax Fundamentals - Exercise/vacation.js
function vacation(amount, type, day) {

    let totalPrice = 0;
    switch (day) {
        case 'Friday':
            if (type === 'Students') {
                totalPrice = 8.45 * amount;
                if (amount >= 30) {
                    totalPrice -= totalPrice * .15;
                }
            } else if (type === 'Business') {
                if (amount >= 100) {
                    amount -= 10;
                }
                totalPrice = 10.90 * amount;

            } else if (type === 'Regular') {
                totalPrice = 15 * amount;
                if (amount >= 10 && amount <= 20) {
                    totalPrice -= totalPrice * 0.05;
                }
            }
            break;
        case 'Saturday':
            if (type === 'Students') {
                totalPrice = 9.80 * amount;
                if (amount >= 30) {
                    totalPrice -= totalPrice * 0.15;
                }
            } else if (type === 'Business') {
                if (amount >= 100) {
                    amount -= 10;
                }
                totalPrice = 15.60 * amount;

            } else if (type === 'Regular') {
                totalPrice = 20 * amount;
                if (amount >= 10 && amount <= 20) {
                    totalPrice -= totalPrice * 0.05;
                }
            }
            break;
        case 'Sunday':
            if (type === 'Students') {
                totalPrice = 10.46 * amount;
                if (amount >= 30) {
                    totalPrice -= totalPrice * 0.15;
                }
            } else if (type === 'Business') {
                if (amount >= 100) {
                    amount -= 10;
                }
                totalPrice = 16 * amount;

            } else if (type === 'Regular') {
                totalPrice = 22.50 * amount;
                if (amount >= 10 && amount <= 20) {
                    totalPrice -= totalPrice * 0.05;
                }
            }
            break;
    }
    console.log(`Total price: ${totalPrice.toFixed(2)}`);

}

vacation(30, "Students", "Friday");
vacation(30, "Students", "Saturday");
vacation(30, "Students", "Sunday");